### DISCUSSION TITLE
Crazyflie no longer connecting to radio

### DISCUSSION THREAD

**Discussion by dwbowen** [bitcraze/discussions#866] [Q&A]
Hello,

I have had the Crazyflie 2.1 and Crazyradio 2.0 running via cfclient and various python scripts successfully on my OSX machine for a while now. But suddenly today the radio no longer connects to the drone with cfclient nor the python script. 

When I click "Connect" in cfclient the green led on the radio comes on but it never connects to the Crazyflie. Or when I run the python script that worked previously the same thing happens: the green led on the radio comes on but it does not connect as before and the script does not run. 

As far as I know I have not made any changes to my system, the Crazyradio or the Crazyflie.

Also, I am unsure if these are related but I have noticed that the on/off button on the Crazyflie no longer functions consistently.

My computer OSX 11.2.5

After having these issues, I attempted to update the Crazyflie firmware to 2023.07 via my iOS. (Somewhat tricky because the power button does not always function properly). At the end of the update, I receive an error ("Error Bootloader connection:") and the back right blue light stays on. And now the Crazyflie does not start up as before. Only the back right blue light. 

Any assistance would be greatly appreciated.

thank you,

**Comment by krichardsson** [comment]
It sounds like you have a half-flashed firmware in the Crazyfie and this is why it does not start up properly any more. 
The way to get it back on track is to do a [recovery firmware flashing](https://www.bitcraze.io/documentation/repository/crazyflie-clients-python/master/userguides/recovery-mode/)

There are a couple of obstacles though:
1. Your power button does not work
2. You have some sort of problem with the radio

How bad is the power button? Does it not work at all or is it a bit glitchy? There might be some ways around this problem, I can look at this tomorrow when I'm back in the office and have access to hardware to test on.

For the non-working radio it is hard to give any good advice. Hardware rarely just stop working and my initial thought is that your system has changed in some way. Did you get an update from Apple or anything else has been updated? 
You mention that the green LED on the radio is lit up which would indicate that the radio is actually doing something and your system is not the problem.
Have you checked that the antenna is properly screwed on and is not lose?

This is a bit of a tricky problem as both your radio and CF is in unknown states. I'll look more at this tomorrow.

**Comment by krichardsson** [comment]
My guess (hope) is that your radio is working and that the Crazyflie has corrupt NRF firmware. Under this assumption the way to proceed is
1. Disconnect the battery on the Crazyflie
2. Press and hold the button and connect the battery again (while still holding the button)
3. Release the button when the blue LEDs start to flash. The Crazyflie should now be in cold boot mode and can be flashed
4. Follow the instructions in the [Recovery firmware flashing](https://www.bitcraze.io/documentation/repository/crazyflie-clients-python/master/userguides/recovery-mode/) instruction, except the "Press and hold the power button on the Crazyflie" part (just skip it) as you already did a different flavor of this above.
